_posts/2022-11-17-what-makes-a-python-package-healthy.md

```diff
@@ -0,0 +1,536 @@
+author: "Leah Wasser"
+permalink: /blog/what-makes-open-source-python-package-healthy
+header:
+    overlay_color: "#666"
+    overlay_filter: 0.6
+categories:
+  - blog-post
+  - highlight
+  - python-packaging
+  - peer-review
+toc: true
+comments: true
+---
+
+
+In previous posts I discussed:
+
+* [Why Python open source software matters to scientists (and package maintainers too)]()
+* [Why collecting metrics surrounding package health matters]()
+
+In this post, I'll summarize a conversation that was held on twitter that gaged 
+what the community thought about metrics to track the health of scientific 
+Python open source packages. 
+
+## Packages and open source software, a few terms to clarify
+
+* When I say Python package or Python open source software, i'm referring to a 
+tool that anyone can install to use in their Python environment. 
+* When i saw open source or free and open source software i'm referring to Python tools that are free to download and and their code is openly available for anyone to see (open source).
+
+## What is package health anyway? 
+
+There are many different ways to think about and evaluate open source 
+Python package health. 
+
+Below is what I posted on Twitter to spur a conversation about what makes a 
+package health. And more specifically what metrics should we (pyOpenSci) collect 
+to evaluate health. 
+
+**My goal:** to see what the community thought about "*what constitutes package health*". 
+
+<blockquote class="twitter-tweet" data-partner="tweetdeck"><p lang="en" dir="ltr">controversial topic: How do we measure the &quot;health&quot; of a <a href="https://twitter.com/hashtag/science?src=hash&amp;ref_src=twsrc%5Etfw">#science</a> <a href="https://twitter.com/hashtag/python?src=hash&amp;ref_src=twsrc%5Etfw">#python</a> package? GitHub stars? downloads, date of latest commit? # of commits a month / quarter? Spread of commits? Thoughts? <a href="https://twitter.com/hashtag/opensource?src=hash&amp;ref_src=twsrc%5Etfw">#opensource</a> <a href="https://twitter.com/hashtag/OpenScience?src=hash&amp;ref_src=twsrc%5Etfw">#OpenScience</a> <a href="https://twitter.com/pyOpenSci?ref_src=twsrc%5Etfw">@pyOpenSci</a></p>&mdash; Leah Wasser 🦉 (@LeahAWasser) <a href="https://twitter.com/LeahAWasser/status/1577730887818498049?ref_src=twsrc%5Etfw">October 5, 2022</a></blockquote>
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+The twitter convo made me realize that there are
+many different perspectives that we can consider when addressing this question. 
+
+More specifically, pyOpenSci is interested in the health of packages that
+support science. So we may need to build upon already existing 
+efforts that determine metrics and customize them to our needs.
+
+<div class="notice" markdown="1">
+### A note about our pyOpenSci packages 
+
+Packages in the pyOpenSci ecosystem are not the large packages like Xarray, 
```

Comment:
Would it be helpful to show the figure from Jake Vanderplas' 2017 pycon talk here
https://www.youtube.com/watch?v=ZyjCqQEUa8o

As referenced here: https://www.coiled.io/blog/pydata-dask

And maybe talk about packages that are "higher up in the stack" or "foundational"?

Reply:
that's a great idea!! perfect image for this. we'd kind of start where astropy is altho astropy is an ecosystem so we are kind of like an intermediate between astropy and scikit  :) i almost need to make a graphic for this i'd use it in presentations too.